libcxx/test/libcxx/strings/basic.string/string.cons/constinit_sso_string.compile.pass.cpp

```diff
@@ -0,0 +1,22 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// UNSUPPORTED: c++03, c++11, c++14, c++17
+
+// Ensure that strings which fit within the SSO size can be constant-initialized
```

Comment:
I think I would also like to see some Sema tests as well, do we expect something like this to work: https://godbolt.org/z/x8EvdGfoq

Currently it does not for libstdc++ but does on MSVC.

```cpp
#include <string>

constexpr std::string f() {
    std::string s = "hello world";

    return s;
}

void g() {
    constexpr auto s = f();
}
```

Reply:
We do expect that to work. It doesn't work on GCC because libstdc++'s std::string has a self-pointer, and a self-pointer is only constant for a global (or function-local static), not a local variable.

Added a test that covers the local var case.